core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java

```diff
@@ -151,20 +151,26 @@ private boolean maySign(
       final Bytes publicKey,
       final Bytes signingRoot,
       final Eth2SigningRequestBody eth2SigningRequestBody) {
+    final Optional<Bytes> gvr =
+        Optional.ofNullable(eth2SigningRequestBody.getForkInfo())
```

Comment:
GVR really can't be optional any more :/

Reply:
ok just read down lower - but - there's a smell here.

Reply:
It was only ever optional for deposits. But since I wanted to pull the gvr field extraction out as common code I made it optional.

Reply:
Gvr is now mandatory and removed the checkArgument. This is required as we cannot get a Eth2SigningRequestBody without a GVR as the ForkInfo is required for block and attestations and gvr is required for ForkInfo so this can never happen.